tests/integration_tests/tests/agentless_tests/test_executions.py

```diff
@@ -603,6 +604,31 @@ def test_cancel_execution_and_then_force_cancel(self):
 
         self._assert_execution_cancelled(execution, deployment_id)
 
+    def test_execute_and_kill_execution(self):
+        """Tests the kill execution option.
+
+        The function executes a new execution, writes the process to a local
+        file, kills the execution and asserts the process was also killed.
+        """
+        execution, node_inst_id, deployment_id = \
+            self._execute_from_resource(workflow_id='read_write',
+                                        workflow_params=None,
+                                        resource_file='dsl'
+                                                      '/read_write_workflows'
+                                                      '.yaml')
+        self._check_state('active_and_wrote', "Execution was expected to go "
+                                              "into 'active_and_wrote' "
+                                              "status", node_inst_id)
+        self._check_state('asleep', "Execution was expected to go"
+                                    " into 'sleeping' status", node_inst_id)
+        content = self.client.node_instances.get(node_inst_id). \
+            runtime_properties['file_content']
+        execution = self.client.executions.cancel(execution.id, force=True,
+                                                  kill=True)
+        expected_status = Execution.KILL_CANCELLING
+        self.assertEquals(expected_status, execution.status)
```

Comment:
How can you be sure the execution is already killed? Maybe add a retry or sleep

Reply:
Added.
        for retry in range(10):
            if self.assertEquals(expected_status, execution.status):
                break
        else:
            raise AssertionError("Process is not killed")